src/utils/ssh.ts

```diff
@@ -0,0 +1,85 @@
+import { exec } from 'child_process';
+
+import * as vscode from 'vscode';
+import { Logger } from '../logger';
+import { ConfigManager } from '../config-manager';
+
+export class SSH {
+  constructor(private readonly configManager?: ConfigManager) {}
+
+  executeCommand(command: string): Promise<string> {
+    Logger.info(command, 'ssh');
+
+    return new Promise((resolve, reject) => {
+      exec(command, (error, stdout) => {
+        if (error) {
+          reject(error);
+        } else {
+          resolve(stdout);
+        }
+      });
+    });
+  }
+
+  async promptForUsername(hostname: string): Promise<string | undefined> {
+    const username = await vscode.window.showInputBox({
+      prompt: `Please enter the username for host "${hostname}"`,
+    });
+
+    if (username && this.configManager) {
+      this.configManager.setUserForHost(hostname, username);
+    }
+
+    return username;
+  }
+
+  async runCommandAndPromptForUsername(hostname: string, command: string) {
```

Comment:
can we add return to type to this function signature? This way you don't need `as string` in the filesystem provider

Reply:
Yup, done.